src/main/scala/io/iohk/ethereum/network/p2p/validators/OmmersValidator.scala

```diff
@@ -0,0 +1,138 @@
+package io.iohk.ethereum.network.p2p.validators
+
+import io.iohk.ethereum.domain.{BlockHeader, Blockchain}
+import io.iohk.ethereum.network.p2p.validators.OmmersValidator.OmmersError._
+
+object OmmersValidator {
+
+  val OmmerGenerationLimit: Int = 6
+  val OmmerSizeLimit: Int = 2
+
+  /**
+    * This method allows validating the ommers of a Block. It performs the following validations (stated on
+    * section 11.1 of http://paper.gavwood.com/):
+    *   - OmmersValidator.validateOmmersLength
+    *   - OmmersValidator.validateOmmersHeaders
+    *   - OmmersValidator.validateOmmersAncestors
+    * It also includes validations mentioned in the white paper (https://github.com/ethereum/wiki/wiki/White-Paper)
+    * and implemented in the different ETC clients:
+    *   - OmmersValidator.validateOmmersNotUsed
+    *   - OmmersValidator.validateDuplicatedOmmers
+    * @param blockNumber    The number of the block to which the ommers belong
+    * @param ommers         The list of ommers to validate
+    * @param blockchain     from where the previous blocks are obtained
+    * @return ommers if valid, an [[OmmersError]] otherwise
+    */
+  def validate(blockNumber: BigInt, ommers: Seq[BlockHeader], blockchain: Blockchain): Either[OmmersError, Seq[BlockHeader]] = {
+    for {
+      _ <- validateOmmersLength(ommers)
+      _ <- validateOmmersHeaders(ommers, blockchain)
+      _ <- validateOmmersAncestors(blockNumber, ommers, blockchain)
+      _ <- validateOmmersNotUsed(blockNumber, ommers, blockchain)
+      _ <- validateDuplicatedOmmers(ommers)
```

Comment:
Maybe this validation can be put before validateOmmersHeaders as it's _les expensive_ and might make validation fail earlier rather than getting ommers from db